lib/CAN/CANPacket.cpp

```diff
@@ -0,0 +1,64 @@
+#include <Arduino.h>
+
+#include "CANPacket.h"
+
+CANPacket::CANPacket() {
+  this->id = 0;
+  this->data = 0;
+}
+
+CANPacket::CANPacket(uint16_t id, uint8_t data[]) {
+  this->id = id;
+
+  for (uint8_t i = 0; i < sizeof(data); i++) {
+    this->data = this->data + (data[i] << (i * 8));
+  }
+}
+
+CANPacket::CANPacket(uint16_t id, uint64_t data) {
+  this->id = id;
+  this->data = data;
+}
+
+void CANPacket::setID(uint16_t id) { this->id = id; }
+
+uint16_t CANPacket::getID() { return this->id; }
+
+void CANPacket::setData(uint8_t data[]) {
+  for (uint8_t i = 0; i < sizeof(data); i++) {
+    this->data = this->data + (data[i] << (i * 8));
+  }
+}
+
+void CANPacket::setData(uint64_t data) { this->data = data; }
+
+uint64_t CANPacket::getData_ui64() { return this->data; }
+
+// ToDo Test unsigned->signed on ESP32 runtime
+int64_t CANPacket::getData_i64() {
+  int64_t result = this->data & 0xFFFFFFFFFFFFFFFF;
+  return result;
+}
+
+uint32_t CANPacket::getData_ui32(uint8_t index) { return (uint32_t)(this->data >> (32 * index)) & 0x00000000FFFFFFFF; }
```

Comment:
Why are uint... and int... handled in same manner? Could simplify and remove and type cast at usage?